src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -1201,7 +1201,13 @@ protected function runGitAction(string $status, GitHub $github, string $provider
 
             $protocol = System::getEnv('_APP_OPTIONS_FORCE_HTTPS') == 'disabled' ? 'http' : 'https';
             $hostname = System::getEnv('_APP_DOMAIN');
-            $providerTargetUrl = "{$protocol}://{$hostname}/console/project-{$project->getId()}/functions/function-{$resource->getId()}";
+
+            $resourceType = match ($resource->getCollection()) {
+                'functions' => 'function',
+                'sites' => 'site',
+                default => throw new \Exception('Invalid resource type')
+            };
+            $providerTargetUrl = "{$protocol}://{$hostname}/console/project-{$project->getId()}/{$resource->getCollection()}/$resourceType-{$resource->getId()}";
```

Comment:
I wouldn't coupld getCollection() with URL.

Let's have enture `$providerTargetUrl` be set by match block. Inside, when we know if its sites or functions, we can hard-code much more